docs/index.md

```diff
@@ -257,6 +257,48 @@ connection "aws_account_z" {
 
 Using named profiles allows Steampipe to work with your existing CLI configurations, including SSO and using role assumption.
 
+### AWS Profile Credentials with IAM Roles
+You may have an IAM User in an AWS account used to hop-off to other accounts via assume role. In this case the AWS Credential files looks like this:
+
+#### aws credential file:
+
+```ini
+[profile_hop_off]
+aws_access_key_id = AKIA4YFAKEKEYXTDS252
+aws_secret_access_key = SH42YMW5p3EThisIsNotRealzTiEUwXN8BOIOF5J8m
+# this IAM User must have permissions to ReadOnlyAccess and sts:AssumeRole for 'arn:aws:iam::*:role/spc_role'
+
+[profile_hop_to_a]
+role_arn = arn:aws:iam::111111111111:role/spc_role
+source_profile = profile_hop_off
+
+[profile_hop_to_b]
+role_arn = arn:aws:iam::222222222222:role/spc_role
+source_profile = profile_hop_off
+```
```

Comment:
```suggestion
```ini
# This profile must have permissions to ReadOnlyAccess and sts:AssumeRole for 'arn:aws:iam::*:role/spc_role'
[profile_hop_off]
aws_access_key_id = AKIA4YFAKEKEYXTDS252
aws_secret_access_key = SH42YMW5p3EThisIsNotRealzTiEUwXN8BOIOF5J8m

[profile_hop_to_a]
role_arn = arn:aws:iam::111111111111:role/spc_role
source_profile = profile_hop_off

[profile_hop_to_b]
role_arn = arn:aws:iam::222222222222:role/spc_role
source_profile = profile_hop_off
```
```

Why does the profile need `ReadOnlyAccess`? I understand `sts:AssumeRole` and `ReadOnlyAccess` is generally considered safe, but I'm curious what your testing has turned up for it.

Reply:
For the AWS docs that I've read, like `https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-role.html#cli-role-overview`, they mention `sts:AssumeRole` being required, but not much about other permissions. Have you tried seeing if the authentication still works even if the IAM user doesn't have `ReadOnlyAccess`?

Reply:
The "hop-off" IAM User can stay without ReadOnlyAcccess Permissions.
But then we need another Profile for an IAM Role (role/spc_role with ReadOnlyAcccess permissions) in the same account, the IAM User resides.
Maybe this is more consistent. 
But the profile "profile_hop_off" may then not be used in aws.spc (as it does not have proper read permissions)

Reply:
@nuvibit-michael Do you know which read permissions the hop-off user needs in order to assume the role? In general, we like to recommend least privilege when we can as a security best practice, so `ReadOnlyAccess` may be required, but it's not clear to me yet why it's required.

For instance, is `ReadOnlyAccess` required because the source profile/user needs to have permissions for the resources that are being queried (and `ReadOnlyAccess` is a broad way of giving all read permissions easily), or do only the role's permissions matter for querying?

Also, when assuming the role, I'm not sure if AWS attempts to list/get the destination role info, since most docs just recommend `sts:AssumeRole`, so it may try to do a direct call to the given role ARN.

Reply:
I observed a redundancy to "AssumeRole Credentials (No MFA)" and allowed myself to merge this with my sample

Reply:
Nice catch, I didn't realize there was a similar section already!